Segmentation fault triggered by timeout

# Summary
Segmentation fault when trying to access storage plugin's GracefulStop method in gocsi before the storage plugin is fully initialized due to timeout.

Is there any parameter we can set in gocsi to avoid this problem?

# Bug Reports
Segmentation fault when trying to access storage plugin's GracefulStop method in gocsi before the storage plugin is fully initialized due to timeout.

We found out about this when investigating the following bug report:
https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/106

## Version
Please include the version of GoCSI and the CSI API model.
GoCSI v1.0.0 and CSI v1.0.0
GoCSI v1.2.0 and CSI v1.2.0

## Expected Behavior
Should log the timeout error but should not have a segmentation fault.

## Actual Behavior
We got a segmentation fault triggered by timeout.

## Steps To Reproduce
This was reported by https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/106. 

I could also reproduce this by adding a wait of 30 seconds and then return error in the following routine which tries to connect to vCenter:
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/vsphere/virtualcenter.go#L195

Then the segmentation fault happened here when it tries to access "sp" which is not fully initialized yet:
https://github.com/rexray/gocsi/blob/master/gocsi.go#L121

## Logs
```
kubectl logs -n kube-system vsphere-csi-controller-0 vsphere-csi-controller 
I1206 07:39:54.069404       1 config.go:261] GetCnsconfig called with cfgPath: /etc/cloud/csi-vsphere.conf
I1206 07:39:54.069647       1 config.go:206] Initializing vc server 160.98.220.50
I1206 07:39:54.069698       1 controller.go:67] Initializing CNS controller
I1206 07:39:54.069734       1 virtualcentermanager.go:63] Initializing defaultVirtualCenterManager...
I1206 07:39:54.069758       1 virtualcentermanager.go:65] Successfully initialized defaultVirtualCenterManager
I1206 07:39:54.069791       1 virtualcentermanager.go:107] Successfully registered VC "160.98.220.50"
I1206 07:39:54.069819       1 manager.go:60] Initializing volume.volumeManager...
I1206 07:39:54.069842       1 manager.go:64] volume.volumeManager initialized
time="2019-12-06T07:40:15Z" level=info msg="received signal; shutting down" signal=terminated
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x128 pc=0x867dc7]

goroutine 24 [running]:
google.golang.org/grpc.(*Server).GracefulStop(0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1393 +0x37
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop.func1()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:333 +0x35
sync.(*Once).Do(0xc0002e080c, 0xc0003eeef8)
        /usr/local/go/src/sync/once.go:44 +0xb3
github.com/rexray/gocsi.(*StoragePlugin).GracefulStop(0xc0002e0780, 0x21183a0, 0xc0000ae010)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:332 +0x56
github.com/rexray/gocsi.Run.func3()
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:121 +0x4e
github.com/rexray/gocsi.trapSignals.func1(0xc000437380, 0xc0004737a0, 0xc000473770)
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:502 +0x143
created by github.com/rexray/gocsi.trapSignals
        /go/pkg/mod/github.com/rexray/gocsi@v1.0.0/gocsi.go:487 +0x107
```

--- a/gocsi.go
+++ b/gocsi.go
@@ -320,7 +320,9 @@ func (sp *StoragePlugin) Serve(ctx context.Context, lis net.Listener) error {
 // errors.
 func (sp *StoragePlugin) Stop(ctx context.Context) {
 	sp.stopOnce.Do(func() {
-		sp.server.Stop()
+		if sp.server != nil {
+			sp.server.Stop()
+		}
 		log.Info("stopped")
 	})
 }
@@ -330,7 +332,9 @@ func (sp *StoragePlugin) Stop(ctx context.Context) {
 // pending RPCs are finished.
 func (sp *StoragePlugin) GracefulStop(ctx context.Context) {
 	sp.stopOnce.Do(func() {
-		sp.server.GracefulStop()
+		if sp.server != nil {
+			sp.server.GracefulStop()
+		}
 		log.Info("gracefully stopped")
 	})
 }

